packages/ripple/src/ripple-directive.ts

```diff
@@ -31,69 +31,91 @@ export interface RippleOptions {
   active?: boolean;
 }
 
-const rippleParts = new WeakMap();
+export interface RippleNodeOptions extends RippleOptions {
+  surfaceNode: HTMLElement;
+}
+
+/**
+ * Applied a ripple to the node specified by {surfaceNode}.
+ * @param options {RippleNodeOptions}
+ */
+export const rippleNode = (options: RippleNodeOptions) => {
+  // TODO(sorvell): This directive requires bringing css yourself. We probably need to do this
+  // because of ShadyCSS, but on Safari, the keyframes styling must be global. Perhaps this
+  // directive could fix that.
+  const surfaceNode = options.surfaceNode;
+  const interactionNode = options.interactionNode || surfaceNode;
+  if (interactionNode.style.position === '') {
+    interactionNode.style.position = 'relative';
+  }
+  const adapter = {
+    browserSupportsCssVars: () => supportsCssVariables,
+    isUnbounded: () =>
+      options.unbounded === undefined ? true : options.unbounded,
+    isSurfaceActive: () => interactionNode![MATCHES](':active'),
+    isSurfaceDisabled: () => Boolean(options.disabled),
+    addClass: (className: string) => surfaceNode.classList.add(className),
+    removeClass: (className: string) =>
+      surfaceNode.classList.remove(className),
+    containsEventTarget: (target: HTMLElement) => interactionNode.contains(target),
+    registerInteractionHandler: (type: string, handler: Handler) =>
+      interactionNode.addEventListener(type, handler, util.applyPassive()),
+    deregisterInteractionHandler: (type: string, handler: Handler) =>
+      interactionNode.removeEventListener(type, handler, util.applyPassive()),
+    registerDocumentInteractionHandler: (evtType: string, handler: Handler) =>
+      document.documentElement.addEventListener(
+          evtType, handler, util.applyPassive()),
+    deregisterDocumentInteractionHandler: (evtType: string, handler: Handler) =>
+      document.documentElement.removeEventListener(
+          evtType, handler, util.applyPassive()),
+    registerResizeHandler: (handler: Handler) =>
+      window.addEventListener('resize', handler),
+    deregisterResizeHandler: (handler: Handler) =>
+      window.removeEventListener('resize', handler),
+    updateCssVariable: (varName: string, value: string) =>
+      surfaceNode.style.setProperty(varName, value),
+    computeBoundingRect: () => interactionNode.getBoundingClientRect(),
+    getWindowPageOffset: () => ({x: window.pageXOffset, y: window.pageYOffset}),
+  };
+  const rippleFoundation = new MDCRippleFoundation(adapter);
```

Comment:
Can you make a Foundation interface for MDCRippleFoundation like in `mwc-switch`?

Reply:
```ts
const rippleFoundation: RippleFoundation = new MDCRippleFoundation(adapter);
```